### PR TITLE
[WIP] Fb web product

### PR DESCRIPTION
**Pull request by K-do**
Refactoring main APP to be able to have main page etc
Adding product management

### CODE REVIEW

File: web/src/App.js [hunk 1]
```diff
@@ -35,18 +45,50 @@ function App() {
...
+                        </div>
+                    }/>
+                    <Route path="/about" exact component={() => <About/>}/>
+                </Switch>
+                <Footer/>
+            </Router>
         </div>
     );
 
+    /* image de l'apocalypse
```
Comment (LeonardBesseau) [inline web/src/App.js:87]: Why are you doing this to me?

File: web/src/pages/About.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                        <img
+                            class="img-fluid rounded mb-4 mb-lg-0"
+                            src="http://placehold.it/900x400"
+                            alt=""
+                        />
+                    </div>
+                    <div class="col-lg-5">
+                        <h1 class="font-weight-light">About</h1>
+                        <p>
+                            Lorem Ipsum is simply dummy text of the printing and typesetting
```
Comment (LeonardBesseau) [inline web/src/pages/About.js:18]: Why not copy paste the git description

File: web/src/pages/Footer.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+function Footer() {
+    return (
+        <div className="footer">
+            <footer class="py-5 bg-dark fixed-bottom">
+                <div class="container">
+                    <p class="m-0 text-center text-white">
+                        Copyright &copy; Your Website 2020
```
Comment (LeonardBesseau) [inline web/src/pages/Footer.js:9]: Please change to 2021

File: web/src/product/ProductForm.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+const ProductForm = (props) => {
+    const formik = useFormik({
+        initialValues: {
+            name: '',
+            tag: ''
+        },
+        onSubmit: async (values) => {
+            const res = await addProduct(props.token, "http://localhost:4000/api/v1/products/add", values);
```
Comment (LeonardBesseau) [inline web/src/product/ProductForm.js:31]: Please use the URL.js file as in #37

**Review by LeonardBesseau — CHANGES_REQUESTED** [review]
Do we really need to use the router to navigate between our pages ?

File: web/src/product/ProductForm.js [hunk 5]
```diff
@@ -21,50 +21,223 @@ async function addProduct(token, url, data) {
...
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+
 const ProductForm = (props) => {
-    const formik = useFormik({
+    const formikAdd = useFormik({
         initialValues: {
```
Comment (LeonardBesseau) [inline web/src/product/ProductForm.js:89]: Might be interesting to take a look at :
https://medium.com/@nphivu414/build-a-multi-step-form-with-react-hooks-formik-yup-and-materialui-fa4f73545598
to make only one form. 
First step, choose which action to take and then display one form or the other.

File: web/src/product/ProductForm.js [hunk 6]
```diff
@@ -21,50 +21,223 @@ async function addProduct(token, url, data) {
...
-                formik.errors.submit = res.message;
+                formikAdd.errors.submit = res.message;
             } else {
                 console.log("Is ok");
+                alert("The product: " + JSON.stringify(res.data.name) + " has correctly been added");
             }
         }
     });
 
+    const formikGet = useFormik({
```
Comment (LeonardBesseau) [inline web/src/product/ProductForm.js:104]: Do we really expect user to know the id of a product ? Not sure this is useful.
Better to have a list that display all products

File: web/src/product/ProductForm.js [hunk 7]
```diff
@@ -21,50 +21,223 @@ async function addProduct(token, url, data) {
...
+        return response.json();
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+
 const ProductForm = (props) => {
-    const formik = useFormik({
+    const formikAdd = useFormik({
```
Comment (LeonardBesseau) [inline web/src/product/ProductForm.js:88]: Maybe move the form into it's own file to improve readability

**Review by LeonardBesseau — CHANGES_REQUESTED** [review]
Welcome, logout and menu seems OK.
## Product
Product needs refactoring.  Separate into different files if possible.
Try to streamline the process as much as possible. Clearly distinguish each use case.
Maybe implement a listSelector (or kindly ask for one)

File: web/src/product/ProductForm.js [hunk 8]
```diff
@@ -21,50 +21,223 @@ async function addProduct(token, url, data) {
...
+            referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
+        });
+        return response.json();
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+async function getAllProduct() {
+    const url = "http://localhost:4000/api/v1/products/";
```
[the remark this answers is not in the document]
Comment (K-do) [reply web/src/product/ProductForm.js:45]: All Product will be refactor, this was just a sketch

File: web/src/product/ProductForm.js (hunk 6 above)
Comment (K-do) [reply web/src/product/ProductForm.js:104]: true

File: web/src/product/ProductForm.js [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
+async function addProduct(url, data) {
     try {
         const response = await fetch(url, {
             method: 'POST', // *GET, POST, PUT, DELETE, etc.
             mode: 'cors', // no-cors, *cors, same-origin
             cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
             credentials: 'same-origin', // include, *same-origin, omit
             headers: {
-                'Authorization': 'Bearer ' + token,
+                'Authorization': 'Bearer ' + localStorage.token,
```
[the remark this answers is not in the document]
Comment (K-do) [reply web/src/product/ProductForm.js:11]: will use props

File: web/src/product/ProductForm.js [hunk 10]
```diff
@@ -21,50 +21,223 @@ async function addProduct(token, url, data) {
...
+            if (res.status === "fail") {
+                formikGet.errors.submit = res.message;
+            } else {
+                console.log("Is ok");
+                alert("This is all the products:\n" + JSON.stringify(res.data));
+            }
+        }
+    });
+
+    const formikUpdate = useFormik({
```
[the remark this answers is not in the document]
Comment (K-do) [reply web/src/product/ProductForm.js:134]: will do

File: web/src/product/ProductForm.js (hunk 7 above)
Comment (K-do) [reply web/src/product/ProductForm.js:88]: Product will be refactor, utils wil probably be added in a separate file

File: web/src/product/ProductForm.js (hunk 5 above)
Comment (K-do) [reply web/src/product/ProductForm.js:89]: I will check this

**Comment by K-do** [comment]
BUG: if we `f5` in browser during sessions, all the requests will have `jwt token malformed`, can't figure out why for now

**Comment by K-do** [comment]
I stop here for this itteration

**Review by LeonardBesseau — APPROVED** [review]
Looks good for me

**Pull request merged**